Samples/NetworkMediaStreamer/RTSPServer/src/RtspSession.cpp

```diff
@@ -0,0 +1,575 @@
+    Response = "RTSP/1.0 200 OK\r\nCSeq:" + m_CSeq + "\r\n"
+        + DateHeader() + "\r\n"
+        + "Range: npt=0.000-\r\n"
+        + "Session: " + std::to_string(m_RtspSessionID) + "\r\n"
+        + "RTP-Info: url=" + m_URLProto + "://" + m_URLHostPort + "\r\n\r\n",
+
+        SendToClient(Response);
+    StopIfStreaming();
+
+    m_pLoggerEvents[(int)LoggerType::RTSPMSGS](S_OK, winrt::to_hstring(__FUNCTION__) + L":Response:" + winrt::to_hstring(Response));
+
+}
+
+void RTSPSession::SendToClient(std::string Response)
+{
+    m_pRtspClient->Send((BYTE*)Response.c_str(), (int)Response.length());
+}
+
+char const* RTSPSession::DateHeader()
+{
+    static char buf[RTSP_PARAM_STRING_MAX];
+    time_t tt = time(NULL);
+    tm t;
+    gmtime_s(&t, &tt);
+    strftime(buf, sizeof buf, "Date: %a, %b %d %Y %H:%M:%S GMT", &t);
+    return buf;
+}
+
+int RTSPSession::GetStreamID()
+{
+    return 0;//m_StreamID;
+}
+
+void RTSPSession::BeginSession(winrt::delegate<RTSPSession*> completed)
+{
+    m_Completed = completed;
+    m_RtspReadEvent.attach(WSACreateEvent());      // create READ wait event for our RTSP client socket
+    WSAEventSelect(m_pRtspClient->GetSocket(), m_RtspReadEvent.get(), FD_READ | FD_CLOSE);   // select socket read event
+    RegisterWaitForSingleObject(m_callBackHandle.put(), m_RtspReadEvent.get(), [](PVOID arg, BOOLEAN flag)
+        {
+            auto pSession = (RTSPSession*)arg;
+            try
+            {
+                WSAResetEvent(pSession->m_RtspReadEvent.get());
+                auto l = std::lock_guard(pSession->m_readDelegateMutex);
+                if (!pSession->m_callBackHandle)
+                {
+                    //Session stopped
+                    return;
+                }
+
+                char* pRecvBuf = (char*)pSession->m_pTcpRxBuff.get();
+
+                RTSP_CMD C = RTSP_CMD::UNKNOWN;
+                int res = pSession->m_pRtspClient->Recv((BYTE*)pRecvBuf, RTSP_BUFFER_SIZE);
+                if (res > 0)
+                {
+                    // we filter away everything which seems not to be an RTSP command: O-ption, D-escribe, S-etup, P-lay, T-eardown
+                    if ((pRecvBuf[0] == 'O') || (pRecvBuf[0] == 'D') || (pRecvBuf[0] == 'S') || (pRecvBuf[0] == 'P') || (pRecvBuf[0] == 'T'))
+                    {
+                        C = pSession->HandleRequest(pRecvBuf, res);
+                    }
+                    else
+                    {
+                        std::ostringstream logstring;
+                        logstring << "\nUnhandled Request ignored : size =" << res << "\nDump:";
+                        for (int i = 0; i < res; i++)
+                        {
+                            logstring << " 0x" << std::hex << ((uint32_t)(uint8_t)pRecvBuf[i]);
+                        }
+                        pSession->m_pLoggerEvents[(int)LoggerType::WARNINGS](S_OK, winrt::to_hstring(logstring.str()));
+                    }

```

Comment:
suggestion: have this done in HandleRequest (a public virtual member func you mean for other to inherit/override) and have it return a RTSP_CMD::UNKNOWN; in case of unknown command